clash-prelude/src/Clash/Signal/BiSignal.hs

```diff
@@ -167,19 +167,15 @@ newtype BiSignalOut (ds :: BiSignalDefault) (dom :: Domain) (n :: Nat)
 type instance HasDomain dom1 (BiSignalOut ds dom2 n) = DomEq dom1 dom2
 type instance TryDomain t (BiSignalOut ds dom n) = 'Found dom
 
-#if MIN_VERSION_base(4,11,0)
 instance Semigroup (BiSignalOut defaultState dom n) where
   (BiSignalOut b1) <> (BiSignalOut b2) = BiSignalOut (b1 ++ b2)
-#endif
 
 -- | Monoid instance to support concatenating
 --
 -- __NB__ Not synthesizable
 instance Monoid (BiSignalOut defaultState dom n) where
   mempty                                    = BiSignalOut []
-#if !MIN_VERSION_base(4,11,0)
   mappend (BiSignalOut b1) (BiSignalOut b2) = BiSignalOut $ b1 ++ b2
```

Comment:
> NOTE: This method is redundant and has the default implementation mappend = '(<>)' since base-4.11.0.0.

Reply:
It _should_ also be an error (pragmatically speaking). Do we not use `-Wcompat`?

Reply:
Now we do! https://github.com/clash-lang/clash-compiler/pull/1210